test/cpp_extensions/complex_registration_extension.cpp

```diff
@@ -39,17 +39,35 @@ struct CPUComplexFloatType : public at::CPUTypeDefault {
   TypeID ID() const override;
 
   Tensor empty(IntArrayRef size, const TensorOptions & options) const override {
-    // Delegate to the appropriate cpu tensor factory
-    const DeviceGuard device_guard(options.device());
-    return at::native::empty_cpu(/* actuals */ size, options);
+    AT_ASSERT(options.device().is_cpu());
+
+    for (auto x: size) {
+      AT_CHECK(x >= 0, "Trying to create tensor with negative dimension ", x, ": ", size);
```

Comment:
Umm... x isn't the dimension index, it's the size of the dimension. You can probably just drop `x` entirely; the printing of `size` is good enough